tests/conftest.py

```diff
@@ -578,3 +584,49 @@ def project_context(project: str | Path, in_place: bool = False) -> Iterator[Pat
             yield path
 
     return project_context
+
+
+@pytest.fixture()
+def temp_repo(tmp_path: Path) -> TempRepoFixture:
+    """Temporary repository with 2 commits"""
+    repo = dulwich.repo.Repo.init(str(tmp_path))
+
+    # init commit
+    (tmp_path / "foo").write_text("foo", encoding="utf-8")
+    repo.stage(["foo"])
+
+    init_commit = repo.do_commit(
+        committer=b"User <user@example.com>",
+        author=b"User <user@example.com>",
+        message=b"init",
+        no_verify=True,
+    )
+
+    # one commit which is not "head
+    (tmp_path / "bar").write_text("bar", encoding="utf-8")
+    repo.stage(["bar"])
+    middle_commit = repo.do_commit(
+        committer=b"User <user@example.com>",
+        author=b"User <user@example.com>",
+        message=b"extra",
+        no_verify=True,
+    )
+
+    # extra commit
+    (tmp_path / "foo").write_text("foo", encoding="utf-8")
```

Comment:
Is this an empty commit because it writes the same as the first commit? (It probably does not matter, I am just wondering.)

Reply:
Thanks for flagging this, I added a new file instead